File: src/minecraft/monnef/jaffas/technic/block/BlockDirectionalTechnic.java
/*
 * Jaffas and more!
 * author: monnef
 */

package monnef.jaffas.technic.block;

import monnef.jaffas.food.block.BlockJDirectional;
import monnef.jaffas.technic.JaffasTechnic;
import monnef.jaffas.technic.Reference;
import net.minecraft.block.material.Material;

public class BlockDirectionalTechnic extends BlockJDirectional {
    public BlockDirectionalTechnic(int id, int textureStart, int texturesCountPerSet, Material material, TextureMappingType type) {
        this(id, textureStart, texturesCountPerSet, material, type, 1);
    }

    public BlockDirectionalTechnic(int id, int textureStart, int texturesCountPerSet, Material material, TextureMappingType type, int textureSetsCount) {
        super(id, textureStart, texturesCountPerSet, material, type, textureSetsCount);
        setCreativeTab(JaffasTechnic.instance.creativeTab);
    }

    @Override
    public String getModName() {
        return Reference.ModName;
    }

    @Override
    public int getDefaultSheetNumber() {
        return 3;
    }
}

File: src/main/scala/monnef/jaffas/technic/block/BlockDirectionalTechnic.scala
package monnef.jaffas.technic.block

import monnef.jaffas.food.block.BlockJDirectional
import monnef.jaffas.technic.JaffasTechnic
import net.minecraft.block.material.Material
import monnef.jaffas.technic.common.IconDescriptorTechnic

class BlockDirectionalTechnic(_textureStart: Int, _texturesCountPerSet: Int, _material: Material, _type: BlockJDirectional.TextureMappingType, _textureSetsCount: Int) extends
BlockJDirectional(_textureStart, _texturesCountPerSet, _material, _type, _textureSetsCount) with IconDescriptorTechnic {
  setCreativeTab(JaffasTechnic.instance.creativeTab)

  def this(id: Int, textureStart: Int, texturesCountPerSet: Int, material: Material, `type`: BlockJDirectional.TextureMappingType) {
    this(id, textureStart, texturesCountPerSet, material, `type`, 1)
  }
}